keras_nlp/samplers/beam_sampler.py

```diff
@@ -186,16 +187,16 @@ def body(prompt, cache, index, log_probs):
 
             def gather_beams(x):
                 x = unflatten_beams(x)
-                x = tf.gather(x, beam_indices, axis=1, batch_dims=1)
+                x = ops.take_along_axis(x, beam_indices, axis=1, batch_dims=1)
```

Comment:
`batch_dims` is not a argument to `take_along_axis`

Reply:
I think this line will throw dimension mismatch between beam_indicea and x